zeroex/order.go

```diff
@@ -298,7 +325,7 @@ type signedOrderJSON struct {
 }
 
 // MarshalJSON implements a custom JSON marshaller for the SignedOrder type
-func (s *SignedOrder) MarshalJSON() ([]byte, error) {
+func (s SignedOrder) MarshalJSON() ([]byte, error) {
```

Comment:
Non-pointer receiver types are safer for `MarshalJSON`. Without this, passing a `SignedOrder` (not a pointer) to `json.Marshal` will happily work but it will *not* use our custom method.